nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -352,12 +352,15 @@ in {
       };
       dbhost = mkOption {
         type = types.nullOr types.str;
-        default = "localhost";
+        default = if cfg.database.createLocally then (if cfg.config.dbtype == "pgsql" then "/run/postgresql" else "localhost") else null;
```

Comment:
we should add a TODO to unify everything behind UNIX socket auth if we use local dbs.